core/modules/layout/layout.admin.inc

```diff
@@ -2449,27 +2478,123 @@ function layout_delete_form($form, &$form_state, $layout) {
+      if ($removal_removes_page) {
+        layout_warn_if_site_info_path($path, 'delete');
+      }
+      break;
 
-  if ($layout->storage === LAYOUT_STORAGE_NORMAL) {
-    $layout->delete();
-    backdrop_set_message(t('The layout has been deleted.'));
-  }
-  else {
-    $layout->revert();
-    backdrop_set_message(t('The layout has been reverted.'));
+    case t('Disable layout'):
+      $layout->disable();
+      backdrop_set_message(t('Layout %title has been disabled.', array('%title' => $title)));
+      if ($removal_removes_page) {
+        layout_warn_if_site_info_path($path, 'disable');
+      }
+      break;
+
+    case t('Revert layout'):
+      $layout->revert();
+      backdrop_set_message(t('Layout %title has been reverted.', array('%title' => $title)));
+      break;
   }
+
   $form_state['redirect'] = 'admin/structure/layouts';
 }
 
+/**
+ * Returns whether the deletion or disabling of this layout would remove the
+ * page at its path, i.e., if the layout creates a page and is the last enabled
+ * layout with its path.
+ */
+function layout_removal_removes_page($layout) {
+  $path = $layout->getPath();
+  $form['layout_path'] = $path;
+  $router_item = menu_get_item($path);
+  if ($router_item != FALSE) {
+    $is_layout_page = ($router_item['page_callback'] == 'layout_page_callback');
+    if ($is_layout_page) {
+      if (!isset($router_item['map'])) {
+        // This needs to exist but isn't always set by menu_get_item().
+        $router_item['map'] = array();
+      }
+      return count(layout_load_multiple_by_router_item($router_item)) == 1;
+    }
+  }
+  return FALSE;
+}
+
+/**
+ * Issues warnings if for a layout whose removal removes a page, its path
+ * matches one of the paths on the site information page.
+ *
+ * @param string $path
+ *   The path of the layout (that was just deleted or disabled).
+ * @param string $op
+ *    One of 'delete' or 'disable'.
+ */
+function layout_warn_if_site_info_path($path, $op) {
+  $site_config = config('system.core');
+  switch ($op) {
+    case 'delete':
+      if ($path == $site_config->get('site_frontpage')) {
+        $url = url('admin/config/system/site-information') . '#edit-front-page';
+        backdrop_set_message(t('You have deleted the page at path %path that is used as the home page of the site. You must change the home page on the <a href="!url">site information configuration page</a> to a different path.', array('%path' => $path, '!url' => $url)), 'warning');
+      }
+      if ($path == $site_config->get('site_403')) {
+        $url = url('admin/config/system/site-information') . '#edit-error-page';
+        backdrop_set_message(t('You have deleted the page at path %path that is used as the default 403 (access denied) page of the site. You must change the 403 page on the <a href="!url">site information configuration page</a> to a different path.', array('%path' => $path, '!url' => $url)), 'warning');
```

Comment:
`You have deleted the page at %path that was used as the default 403 (access denied) page for this site. You should also change the 403 page setting under <a href="!url">site information</a>.`

Reply:
Changed.